tchannel-core/src/main/java/com/uber/tchannel/tracing/RequestSpanInterceptor.java

```diff
@@ -0,0 +1,31 @@
+package com.uber.tchannel.tracing;
+
+import com.uber.tchannel.messages.Request;
+import io.opentracing.Span;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * A {@link TracingContext} implementation decorated with this interface will get an option to intercept an inbound
+ * and/or outgoing request right after the tracing {@link Span} has been created and populated.
+ */
+public interface RequestSpanInterceptor {
```

Comment:
you may want to split this into InboundInterceptor and OutboundInterceptor. Otherwise it enforces unnecessary coupling on the implementation.

Reply:
I contemplated the dilemma and couldn't decide either way, so I used the simpler approach with fewer interfaces / moving parts. If you have a strong opinion/arguments towards the separate interfaces - I'll be happy to switch. The only use case I have in mind is utilizing both methods.